types/index.d.ts

```diff
@@ -0,0 +1,54 @@
+// TypeScript Version: 3.0
+
+import {Node, Parent} from 'unist'
+import {Test} from 'unist-util-is'
+
+declare namespace mdastUtilToc {
+  interface Link extends Parent {
+    type: 'link'
+    title: null
+    url: string
+    children: Node[]
+  }
+
+  interface Paragraph extends Parent {
+    type: 'paragraph'
+    children: Link[]
+  }
+
+  interface ListItem extends Parent {
+    type: 'listItem'
+    loose: boolean
+    spread: boolean
+    children: Node[]
+  }
+
+  interface List extends Parent {
+    type: 'list'
+    ordered: boolean
+    spread: boolean
+    children: ListItem[]
+  }
+
+  interface TOCOptions {
+    heading?: string
+    maxDepth?: number
+    skip?: string
+    tight?: boolean
+    prefix?: string
+    parents?: Test<Node> | Array<Test<any>>
```

Comment:
I'm not sure about this.. Took it from the type definition of `unist-util-is`

https://github.com/syntax-tree/unist-util-is/blob/master/index.d.ts#L60

But I wasn't sure how to deal with the Type parameter

Reply:
Looks good, in this case is should be safe to make this a bit stricter and set

```suggestion
    parents?: Test<Node> | Array<Test<Node>>
```